repositories/implementations/csv_mlb_player.go

```diff
@@ -87,3 +89,98 @@ func (repo *CSVMLBPlayerRepository) GetMLBPlayerByID(id int) (*e.MLBPlayer, erro
 
 	return nil, nil
 }
+
+func (repo *CSVMLBPlayerRepository) GetMLBPlayerDesired(filterType string, totalItems int, itemsPerWorker int) ([]e.MLBPlayer, error) {
+	f, err := os.Open(repo.filePath)
+
+	if err != nil {
+		log.Println("error opening the file", err)
+		return nil, errors.New("error opening the file")
+	}
+	defer f.Close()
+
+	reader := csv.NewReader(f)
+	reader.Read()
+
+	pCount := 0
+	job := func(jobID int) (*e.MLBPlayer, bool) {
+		pCount++
+		data, err := reader.Read()
+
+		if err == io.EOF {
+			log.Println("JOB", jobID, "reached end of file")
+
+			return nil, true
+		}
+
+		if err != nil {
+			log.Println("JOB", jobID, "error reading line: ", err)
+
+			return nil, false
+		}
+		id, err := strconv.Atoi(data[0])
+
+		if err != nil {
+			return nil, false
+		}
+		height, err := strconv.Atoi(data[4])
+
+		if err != nil {
+			return nil, false
+		}
+		weight, err := strconv.ParseFloat(data[5], 32)
+		if err != nil {
+			return nil, false
+		}
+		age, err := strconv.ParseFloat(data[6], 32)
+
+		if err != nil {
+			return nil, false
+		}
+
+		return &e.MLBPlayer{
+			ID:       id,
+			Name:     data[1],
+			Team:     data[2],
+			Position: data[3],
+			Height:   height,
+			Weight:   float32(weight),
+			Age:      float32(age),
+		}, false
+	}
+
+	jobs := make(chan int, totalItems)
+	results := make(chan *e.MLBPlayer, totalItems)
+	workersCount := totalItems / itemsPerWorker
+
+	log.Println(workersCount)
+	for i := 0; i < workersCount; i++ {
+		go func(wokerID int, jobs <-chan int, results chan<- *e.MLBPlayer) {
+			for j := range jobs {
+				log.Println("Worker", wokerID, "executed job", j)
+				p, f := job(j)
+				if f {
+					close(results)
+					break
+				}
+				results <- p
+				if pCount == totalItems {
```

Comment:
I don't think this can guarantee that each worker will process the exact amount of items it should. Maybe a thread ended up processing more or less than it should, you would like to have a counter per worker to make sure each process the amounts of items it should